ingester/src/querier_handler.rs

```diff
@@ -1,74 +1,231 @@
     ))]
     ExecutePlan { source: DataFusionError },
+
+    #[snafu(display("Error collecting a stream of record bacthes: {}", source))]
+    CollectStream { source: DataFusionError },
+
+    #[snafu(display(
+        "No Table Data found for the given sequencer id {}, namspace name {}, table name {}", sequencer_id.get(), namespace_name, table_name
+    ))]
+    TableNotFound {
+        sequencer_id: SequencerId,
+        namespace_name: String,
+        table_name: String,
+    },
+
+    #[snafu(display("Error snapshotting non-persisting data: {}", source))]
+    SnapshotNonPersistingData { source: data::Error },
+
+    #[snafu(display("Error concating same-schema record bacthes: {}", source))]
+    ConcatBatches { source: arrow::error::ArrowError },
 }
 
 /// A specialized `Error` for Ingester's Query errors
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
 /// Return data to send as a response back to the Querier per its request
 pub async fn prepare_data_to_querier(
-    _request: &IngesterQueryRequest,
+    ingest_data: Arc<IngesterData>,
+    request: &IngesterQueryRequest,
 ) -> Result<IngesterQueryResponse> {
-    // Steps:
-    //  1. Read the IngesterData to get TableData for the given request's table
-    //  3. For each PartitionData of the TableData above, get its DataBuffer by:
-    //     3.1. Snapshot the buffer by invoking DataBuffer::snapshot
-    //          This snapshot will not inlcude any tombstones because if there are tomstones
-    //          right before this, it would trigger the sanpshot creation already
-    //     3.2. Converting all snapshots to a QueryableBatch that won't have any tombstones.
-    //     3.3. Get PersistingBatch's QueryableBatches and add tomstones in the list `deletes_during_persisting` if any
-    //     3.4. For each QueryableBatch produced in 3.2 and 3.3, invoke the avaialble `query` function to filter data per request.
-    //          . The request's columns, time range, and predicates will be applied at this step. Each will return a RecordBatch.
-    //     3.5. Put all RecordBatches in a stream
-    //  4. The stream is a part of the returned result
-
-    let schema = Arc::new(ArrowSchema::new(vec![]));
-    let stream = datafusion::physical_plan::EmptyRecordBatchStream::new(Arc::clone(&schema));
+    // ------------------------------------------------
+    //  Read the IngesterData to get TableData for the given request's table
+    let table_data = ingest_data.table_data(
+        request.sequencer_id,
+        request.namespace.as_str(),
+        request.table.as_str(),
+    );
+    if table_data.is_none() {
+        return Err(Error::TableNotFound {
+            sequencer_id: request.sequencer_id,
+            namespace_name: request.namespace.clone(),
+            table_name: request.table.clone(),
+        });
+    }
+
+    // ------------------------------------------------
+    // Accumulate data from each partition
+    let table_data = table_data.unwrap();
```

Comment:
This `unwrap`, even though it's ok because of the `return` if `table_data` is none above, takes a minute to read and verify that this is ok and won't actually panic. I think this would be clearer if we add `use snafu::OptionExt;` at the top and then do:

```suggestion
    let table_data = table_data.context(TableNotFoundSnafu {
        sequencer_id: request.sequencer_id,
        namespace_name: &request.namespace,
        table_name: &request.table,
    })?;

    // ------------------------------------------------
    // Accumulate data from each partition
```

This also lets snafu take care of cloning when needed.